src/ui/camera.js

```diff
@@ -1215,9 +1213,49 @@ class Camera extends Evented {
                 tr.setLocationAtPoint(tr.renderWorldCopies ? newCenter.wrap() : newCenter, pointAtOffset);
             }
 
-            this._fireMoveEvents(eventData);
+            if (!options.preloadOnly) {
+                this._fireMoveEvents(eventData);
+            }
+
+            return tr.clone();
+        };
+
+        if (options.preloadOnly) {
+            const framerateTarget = 15;
+            const frameTimeTarget = 1000 / (framerateTarget * options.duration);
+            const emulateFrame = frame(tr.clone());
+
+            const predictedTransforms = [];
+            for (let i = 0; i <= 1; i += frameTimeTarget) {
+                const transform = emulateFrame(i);
+                predictedTransforms.push(transform);
+            }
 
-        }, (interruptingEaseId?: string) => {
+            this._preloadTiles(predictedTransforms);
+
+            return;
+        }
```

Comment:
Could this and [this block](https://github.com/mapbox/mapbox-gl-js/pull/11328/files?authenticity_token=rJuiKYA0u7BuO3mJDh2hSRAkZvtdPu%2FKVHMz%2F4Ug0I7NvbcIHSDK4XY4DJsmDcH7MPphUHV5le3QqCmttsQF3A%3D%3D&file-filters%5B%5D=.js#diff-6e31e9c70d2c897b3a306e5a6718a4b6698e8b5eb3a11729da97fe21ccc45261R1543) be wrapped in a function for brevity? And also so that the fixed framerate configuration is in the same place.